.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,95 @@
+name: "CI/CD"
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+      - release/*
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: "Publish on expo.dev"
+    uses: ./.github/workflows/publish.yml
+
+  build-android:
+    name: "Build Android APK"
+    uses: ./.github/workflows/build-android.yml
+
+  build-ios:
+    name: "Build iOS App"
+    uses: ./.github/workflows/build-ios.yml
+
+  check-release-version:
+    name: "Check release version"
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/release/')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v2
+        with:
+          # actions/checkout does a shallow clone by default, which doesn't pull any tags
+          fetch-depth: 0
+
+      - name: Get current package version
+        id: extract_version
+        uses: Saionaro/extract-package-version@v1.0.6
+
+      - name: Check that package version and release version match
+        if: startsWith(github.ref, 'refs/heads/release/')
+        run: |
+          ref_version="${GITHUB_REF#refs/heads/release/}"
+          if [[ "$ref_version" != "${{ steps.extract_version.outputs.version }}" ]]; then
+            echo "::error title=Version Mismatch::Version '$ref_version' from branch name does not match package version '${{ steps.extract_version.outputs.version }}'. Did you bump the package version?"
+            exit 1
+          fi
+
+      - name: Check that no tag for the current version already exists
+        run: |
+          git tag -l
+          if git tag -l | grep -Fxq "v${{ steps.extract_version.outputs.version }}"; then
+            echo "::error title=v${{ steps.extract_version.outputs.version }} already exists::The release tag 'v${{ steps.extract_version.outputs.version }}' already exists."
+            exit 1
+          fi
+
+  release:
+    name: "Release"
+    if: github.ref == 'refs/heads/main'
+    needs: [ publish, build-android, build-ios, check-release-version ]
+    runs-on: ubuntu-latest
+    # Prevent multiple release jobs running at once
+    concurrency: release
+    permissions:
+      # Needed to push tags
+      contents: write
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: fregante/setup-git-user@v1
+
+      - uses: actions/download-artifact@v3
+        with:
+          path: artifacts
```

Comment:
Werden hier alle artifacts des Repos geladen oder wie werden die spezifiziert?

Reply:
Artefakte gehören immer zu dem Workflow run, in dem sie hochgeladen wurden. Hier werden alle Artefakte dieses Workflow runs heruntergeladen, auch, wenn das in der Doku nicht eindeutig ist: https://github.com/actions/download-artifact#download-all-artifacts

Reply:
Ah okay. Ich bin davon ausgegangen, dass weil die workflows aus separaten Dateien aufgerufen werden, deren artifacts nicht in diesem workflow zugreifbar sind. ex: `uses: ./.github/workflows/build-android.yml`.